docs/contrib/contributing.rst

```diff
@@ -89,18 +89,32 @@ In order for a Pull Request to be considered, it has to meet these requirements:
 
 2.  Be accompanied by appropriate tests.
 
-3.  Be issued from a branch *other than master* (PRs coming from master will not
+3.  Be accompanied by appropriate documentation
+    :ref:`documentation <project-documentation>`.
+
+4.  Be issued from a branch *other than master* (PRs coming from master will not
     be accepted.)
 
 If not *all* of these requirements are met then the code should **not** be
 merged into the distribution, and need not even be reviewed.
 
+
+.. _project-documentation:
+
 Documentation
 -------------
 
-All code contributed to the user-facing standard library (the `nativelib/`
-directory) should come accompanied with documentation.
-Pull requests containing undocumented code will not be accepted.
+Pull requests, other than the most trivial, without documentation
+will not be accepted. Imagine yourself as the person merging trying
+to understand, use, merge, or maintain the contribution.
+
+All code containing changes
+potentially visible to end users or developers contributed to user-facing
+standard libraries (nativelib, clib, javalib, etc.) should come accompanied
```

Comment:
Changes to the javalib don't require documentation, and typically should *not* have documentation, because there is no better doc for them than the JavaDoc, and we can't copy the JavaDoc text.

Reply:
I concur about the value of the official JavaDoc.

For the reasons you describe, javalib.rst has no JavaDoc. It does have two major sections:
"Supported classes" and "Regular expressions (java.util.regex)". Suggesting that 
submitted PRs keep those, and other quirks, up to date is likely to lead to greater
user & developer satisfaction and decreased support effort.

That was my intention/goal.  Should I re-phrase, clarify, or do something otherwise?

Reply:
On re-visiting the code in question to re-base, I believe I have figured out a way to address
this concert.  A footnote now describes how SN javalib doc only describes where it 
varies from the general JDK doc (and gives URLs.)  

The review comment served a good purpose.

Reply:
Concern addressed, I think, by adding a clarifying footnote that SN javalib.rst describes only
places where the implementation differs from the Java API, giving URLs for both.

This should give contributors the clue they need to check javalib.rst and add differences/variances
and that the Java API remains the baseline and need not be documented.